ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs

```diff
@@ -11,39 +11,82 @@
 
 namespace ASPNetCoreMastersTodoList.Api.Controllers
 {
+    [ApiController]
+    [Route("[controller]")]
     public class ItemsController : Controller
     {
         private readonly ILogger<ItemsController> _logger;
-        private ItemService _itemService;
 
-        public ItemsController(ILogger<ItemsController> logger, ItemService itemService)
+        private IItemService _itemService;
+
+        public ItemsController(ILogger<ItemsController> logger, IItemService itemService)
         {
             _logger = logger;
             _itemService = itemService;
         }
 
         [HttpGet]
-        IEnumerable<string> GetAll(int userId)
+        IActionResult Get()
+        {
+            var result = _itemService.GetAll();
+
+            return Ok(result);
+        }
+
+        [HttpGet]
+        [Route("{itemId}")]
+        IActionResult Get(int itemId)
         {
-            return _itemService.GetAll(userId);
+            var result = _itemService.Get(itemId);
+
+            return Ok(result);
         }
 
         [HttpGet]
-        int Get(int userId)
+        [Route("filterBy")]
+        IActionResult GetByFilters([FromBody] string filters)
```

Comment:
parameter filters should be of type `Dictionary<string, string>`

Reply:
Thank you for reviewing  - this was updated.